xml/pci-passthrough

```diff
@@ -0,0 +1,83 @@
+PCI passthrough for SOC-8
+-------------------------
+
+
+There are two parts to leverage pci passthrough in a SOC8 compute node
+
+
+### Part 1
+
+Preparing the compute node
+
+##### check 1:
+There should be no kernel driver and/or binary with direct acces to the PCI device
+and if there are kernel modules they should be blacklisted
+
+for eg: it is common to have the *nouveau* driver installed while the node was installed.
+this driver is a graphics driver for Nvidia based GPU. And it needs to be blacklisted
+
+`echo 'blacklist nouveau' >> /etc/modprobe.d/nouveau-default.conf`
+
+the name of the file is not really important want matters is the location of file and contents
+similarly other drivers can be blacklisted (including Nvidia drivers on the node)
+
+##### Check 2:
+`iommu_groups` should be enanbled on the host, to check if iommu is enabled
+[source,sh]
+```
+node:~/:[0]# virt-host-validate
+.....
+  QEMU: Checking if IOMMU is enabled by kernel   : WARN (IOMMU appears to be disabled in kernel. Add intel_iommu=on to kernel cmdline arguments)                                                                
+.....
+```
+that warning shows if iommu groups are enabled, these groups are needed for passthrough
+
+To modify the kernel cmd line edit the file `/etc/default/grub` and append `intel_iommu=on` to `GRUB_CMDLINE_LINUX_DEFAULT` variable and run `update-bootloader`
+
+A reboot willl be required for the iommu_groups to be created if the the above command showed a warning.
+After the reboot
+
+[source,sh]
+```
+node:~/:[0]# virt-host-validate
+.....
+ QEMU: Checking if IOMMU is enabled by kernel    : PASS
+.....
+```
+
+##### Check 3: (not needed, but just a confirmation step)
+once the iommu_groups are available try to find the group associated with your pci device
+
+eg: for Nvidia gpu
+
+```
+lspci -nn | grep -i nvidia
+08:00.0 VGA compatible controller [0300]: NVIDIA Corporation GT218 [NVS 300] [10de:10d8] (rev a2)
+08:00.1 Audio device [0403]: NVIDIA Corporation High Definition Audio Controller [10de:0be3] (rev a1)
+```
+
+
+here:
+
+`08:00.0`, `08:00.1` are addresses fo the pci device
+
+`10de` is vendorID
+
+`10d8`,`0be3` are productID
+
+
+```
+ls -ld /sys/kernel/iommu_groups/*/devices/*08:00.?/ 
+drwxr-xr-x 3 root root 0 Feb 14 13:05 /sys/kernel/iommu_groups/20/devices/0000:08:00.0/
+drwxr-xr-x 3 root root 0 Feb 19 16:09 /sys/kernel/iommu_groups/20/devices/0000:08:00.1/
+```
+this shows that the devices are available for passthrough
+
+NOTE: when deailing with PCI passthrough a only whole iommu group can be passed and no part of it, here it is group 20.
+
+
+### Part 2
+
+Preparing Nova and Glance for passthrough:
+
+https://docs.openstack.org/nova/rocky/admin/pci-passthrough.html
```

Comment:
Is there anything to be done to make sure that a device is only used by one specific instance (which I hear would be required for GPU)? I don't see any doc about this.

Reply:
once the compute node is configured correctly with `alias` and `pcipassthough_whitelist` it reports count and stats based on whitelist to scheduler, and scheduler does the book-keeping of what is used and what is available. 

that link mentions configuring nova-scheduler
As soon as nova-scheduler uses `PCIPassthoughFilter` it starts filtering nodes which have the required pci device while creating an instance. Since it also keeps track of what is assigned the filter rejects if no nodes with the required pci device are available or all available are already assigned

Reply:
Ah, I missed the link in the link. Which I think may mean that others will miss it... Is there any reason we don't use `PCIPassthoughFilter` by default?

(my concern really is that it's easy to make a mistake while following this doc -- but that's not a blocker for the pull request, it's more about how we deliver the feature)

Reply:
yes, its possible to miss the second link, I will add a note about configuring  both.: scheduler and compute

There are is no reason for that filter to be missed, but there are other filters like CoreFilter and RamFilter which are also required for `*_raio` settings to have an effect.

I will create a PR for PCI passthrough filter